### PR TITLE
Big fix

### PR DESCRIPTION
**Pull request by OsirisAnubis**
1) Added checking of arguments to the function 'check'.                         
2) Removed incorrect checks in the function 'check'.                            
3) Removed function filterByPrice. Added function checkPrice.                   
4) Replaced let to const.                                                       
5) Replaced arrays to array of arrays.                                          
6) Replaced null to Infinity and zero.                                          
7) Changed the output of the results.

### CODE REVIEW

**Comment by OsirisAnubis** [comment]
8) Added semicolons.

File: filterPricesByRange.js [hunk 1]
```diff
@@ -1,41 +1,46 @@
...
+  checkRequirePrice(requireLess, requireGreat, priceLess, priceGreat);
+  if ((priceLess > requireGreat) || (priceGreat < requireLess)) return false;
+  return true;
 }
 
+const checkPrice = (less, great, course) => check(less, great, ...course.prices);
+
 // Task
-let courses = [
+const courses = [
```
Comment (eugeek) [inline filterPricesByRange.js:23]: Tests can't be changed

File: filterPricesByRange.js [hunk 2]
```diff
@@ -1,41 +1,46 @@
...
+  { name: "Courses in USA", prices: [200, Infinity] },
   { name: "Courses in Kazakhstan", prices: [56, 324] },
-  { name: "Courses in France", prices: [null, null] },
+  { name: "Courses in France", prices: [0, Infinity] },
 ];
 
-let requiredRange1 = [null, 200];
-let requiredRange2 = [100, 350];
-let requiredRange3 = [200, null];
+const requiredRanges = [
```
Comment (eugeek) [inline filterPricesByRange.js:34]: Here too

**Review by eugeek — APPROVED** [review]

File: filterPricesByRange.js (hunk 2 above)
Comment (OsirisAnubis) [reply filterPricesByRange.js]: I returned the previous tests.

Comment (OsirisAnubis) [reply filterPricesByRange.js]: Now the function works with null

**Pull request merged**